lib/puppet/application/device.rb

```diff
@@ -169,8 +174,15 @@ def main
     # find device list
     require 'puppet/util/network_device/config'
     devices = Puppet::Util::NetworkDevice::Config.devices
+    if options[:node]
+      devices.select! { |key, value| key == options[:node] }
```

Comment:
so how this works is cruddy, assigning the global list of devices to a local variable is not great 🤢.  But there is only one calling function to self.devices and it is this code.  
So  we can leave it as is. 
OR 
I can rewrite this so the code lives inside lib/puppet/util/network_device/config.rb